### ISSUE TITLE
Execution of Quickstart Tutorial fails because of StorageError

### ISSUE THREAD

**Issue by pawel-kedzior-sw** [theupdateframework/tuf#1087]
**Description of issue**:
After successful installation (according to docs/INSTALLATION.rst), when going through docs/QUICKSTART.md the command

`client.py --repo http://localhost:8001 testfile`

fails with the following output:

```
[root@1f3efbed99bb tufclient]# client.py --repo http://localhost:8001 testfile
127.0.0.1 - - [20/Jul/2020 14:26:52] code 404, message File not found
127.0.0.1 - - [20/Jul/2020 14:26:52] "GET /metadata/2.root.json HTTP/1.1" 404 -
ERROR:root:HTTP error for root version 2
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/tuf/client/updater.py", line 1150, in _update_root_metadata
    version=next_version)
  File "/usr/lib/python2.7/site-packages/tuf/client/updater.py", line 1786, in _update_metadata
    upperbound_filelength, version)
  File "/usr/lib/python2.7/site-packages/tuf/client/updater.py", line 1602, in _get_metadata_file
    raise tuf.exceptions.NoWorkingMirrorError(file_mirror_errors)
NoWorkingMirrorError: No working mirror was found:
  u'localhost:8001': HTTPError(u'404 Client Error: File not found for url: http://localhost:8001/metadata/2.root.json',)
INFO:tuf.download:Downloading: u'http://localhost:8001/metadata/timestamp.json'
127.0.0.1 - - [20/Jul/2020 14:26:52] "GET /metadata/timestamp.json HTTP/1.1" 200 -
INFO:tuf.download:Downloaded 557 bytes out of an upper limit of 16384 bytes.
INFO:tuf.download:Downloading: u'http://localhost:8001/metadata/snapshot.json'
127.0.0.1 - - [20/Jul/2020 14:26:52] "GET /metadata/snapshot.json HTTP/1.1" 200 -
INFO:tuf.download:Downloaded 472 bytes out of an upper limit of 2000000 bytes.
INFO:tuf.download:Downloading: u'http://localhost:8001/metadata/targets.json'
127.0.0.1 - - [20/Jul/2020 14:26:52] "GET /metadata/targets.json HTTP/1.1" 200 -
INFO:tuf.download:Downloaded 735 bytes out of an upper limit of 5000000 bytes.
INFO:tuf.client.updater:u'targets.json' up-to-date.
INFO:tuf.client.updater:u'targets.json' up-to-date.
Traceback (most recent call last):
  File "/usr/bin/client.py", line 237, in <module>
    update_client(arguments)
  File "/usr/bin/client.py", line 139, in update_client
    updated_targets = updater.updated_targets(target_fileinfo, destination_directory)
  File "/usr/lib/python2.7/site-packages/tuf/client/updater.py", line 3180, in updated_targets
    algorithm=algorithm)
  File "/usr/lib/python2.7/site-packages/securesystemslib/hash.py", line 375, in digest_filename
    with storage_backend.get(filename) as file_object:
  File "/usr/lib/python2.7/site-packages/securesystemslib/storage.py", line 205, in __enter__
    "Can't open %s" % self.filepath)
securesystemslib.exceptions.StorageError: Can't open ./tuftargets/testfile
```

**Current behavior**:
client.py fails to download the file

**Expected behavior**:
Go through Quickstart smoothly.

**Notes**:
This reproduced for me on two distributions of Linux
* CentOS Linux release 8.2
* Ubuntu 18.04

**Comment by joshuagl** [comment]
Thanks for reporting this issue @pawel-kedzior-sw! I believe you're running the latest (0.12.2) release of tuf with the latest (0.15.0) release of securesystemslib?

It looks like the storage abstraction we added in securesystemslib 0.15.0 broke the updater code, because securesystemslib now emits a `securesystemslib.exceptions.StorageError` where an `IOError` was previously expected.

This issue was addressed in 0c0aaa97ebb29918f5c9c981b592e97459a5437f, but we have not yet made a release of tuf since merging that change.

If you are comfortable running tuf from HEAD of the develop git branch, you should be able to complete the tutorials as documented.

**Comment by pawel-kedzior-sw** [comment]
Hi @joshuagl , you are correct about the versions which pip installed on my machines.
```
$ pip show tuf
Name: tuf
Version: 0.12.2
(...)
$ pip show securesystemslib
Name: securesystemslib
Version: 0.15.0
(...)
```

**Comment by pawel-kedzior-sw** [comment]
Ok, I was able to work-around the problem by downgrading securesystemslib to 0.14.2.
@joshuagl - many thanks for giving the clue.
```
$ pip uninstall securesystemslib
$ pip install securesystemslib==0.14.2
```
This is acceptable for my learning and demo purposes.

**Comment by joshuagl** [comment]
We have released [tuf 0.13.0](https://github.com/theupdateframework/tuf/releases/tag/v0.13.0), which depends on securesystemslib 0.15.0 or newer. Therefore it should now be possible to follow the tutorial using the packaged versions from PyPI without issue.
